bitnami/ghost/templates/deployment.yaml

```diff
@@ -1,197 +1,189 @@
-              key: mariadb-password
-        {{- else }}
-          value: {{ .Values.externalDatabase.password | quote }}
-        {{- end }}
-        - name: GHOST_HOST
-          value: {{ include "ghost.host" . | quote }}
-        - name: GHOST_PROTOCOL
-          value: {{  .Values.ghostProtocol | quote }}
-        - name: GHOST_PORT_NUMBER
-        {{- if .Values.ghostPort }}
-          value: {{ .Values.ghostPort | quote }}
-        {{- else }}
-          value: {{ .Values.service.port | quote }}
-        {{- end }}
-        - name: GHOST_USERNAME
-          value: {{ .Values.ghostUsername | quote }}
-        - name: GHOST_PASSWORD
-          valueFrom:
-            secretKeyRef:
-              name: {{ template "ghost.fullname" . }}
-              key: ghost-password
-        - name: GHOST_EMAIL
-          value: {{ .Values.ghostEmail | quote }}
-        - name: BLOG_TITLE
-          value: {{ .Values.ghostBlogTitle | quote }}
-        {{- if .Values.smtpHost }}
-        - name: SMTP_HOST
-          value: {{ .Values.smtpHost | quote }}
-        {{- end }}
-        {{- if .Values.smtpPort }}
-        - name: SMTP_PORT
-          value: {{ .Values.smtpPort | quote }}
-        {{- end }}
-        {{- if .Values.smtpUser }}
-        - name: SMTP_USER
-          value: {{ .Values.smtpUser | quote }}
-        {{- end }}
-        {{- if .Values.smtpPassword }}
-        - name: SMTP_PASSWORD
-          valueFrom:
-            secretKeyRef:
-              name: {{ template "ghost.fullname" . }}
-              key: smtp-password
-        {{- end }}
-        {{- if .Values.smtpFromAddress }}
-        - name: SMTP_FROM_ADDRESS
-          value: {{ .Values.smtpFromAddress | quote }}
-        {{- end }}
-        {{- if .Values.smtpService }}
-        - name: SMTP_SERVICE
-          value: {{ .Values.smtpService | quote }}
-        {{- end }}
-        ports:
-        - name: http
-          containerPort: 2368
-        {{- if .Values.livenessProbe.enabled }}
-        livenessProbe:
-          httpGet:
-            path: {{ .Values.ghostPath }}
-            port: http
-            httpHeaders:
-            - name: Host
-              value: {{ include "ghost.host" . | quote }}
-            {{- if eq .Values.ghostProtocol "https" }}
-            - name: X-Forwarded-Proto
-              value: https
+        - name: {{ include "ghost.fullname" . }}
+          image: {{ include "ghost.image" . }}
+          imagePullPolicy: {{ .Values.image.pullPolicy | quote }}
+          env:
+            - name: ALLOW_EMPTY_PASSWORD
+              value: {{ ternary "yes" "no" .Values.allowEmptyPassword | quote }}
+            - name: MARIADB_HOST
+              value: {{ ternary (include "ghost.mariadb.fullname" .) .Values.externalDatabase.host .Values.mariadb.enabled | quote }}
+            - name: MARIADB_PORT_NUMBER
+              value: {{ ternary "3306" .Values.externalDatabase.port .Values.mariadb.enabled | quote }}
```

Comment:
Is the mariadb sub-chart service port configurable? If so, you cannot assume that the sub-chart would use 3306.

Reply:
(very weird case, to be honest)

Reply:
Yes you are right, also I will fix a bug that I found with `ternary`.

```yaml
            {{- $host := .Values.externalDatabase.host -}}
            {{- $port := .Values.externalDatabase.port -}}
            {{- $database := .Values.externalDatabase.database -}}
            {{- $user := .Values.externalDatabase.user -}}

            {{- if .Values.mariadb.enabled -}}
              {{- $host = (include "ghost.mariadb.fullname" .) -}}
              {{- $port = .Values.mariadb.service.port -}}
              {{- $database = .Values.mariadb.db.name -}}
              {{- $user = .Values.mariadb.db.user -}}
            {{- end }}
            - name: MARIADB_HOST
              value: {{ $host | quote }}
            - name: MARIADB_PORT_NUMBER
              value: {{ $port | quote }}
            - name: GHOST_DATABASE_NAME
              value: {{ $database | quote }}
            - name: GHOST_DATABASE_USER
              value: {{ $user | quote }}
```